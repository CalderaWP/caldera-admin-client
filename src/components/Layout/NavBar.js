import React from 'react';
import classNames from 'classnames'
import {NavigableMenu, MenuItem} from '@wordpress/components';
import PropTypes from 'prop-types'

export const NavBar = (props) => {

	return (
		<div
			className={classNames(NavBar.classNames)}
		>
			<NavigableMenu onNavigate={props.onNavigate}>
				{props.choices.map(item => {
					const isSelected = props.value === item.value;
					const icon = isSelected ? 'yes' :
						item.hasOwnProperty('icon') ? item.icon : false;
					return (
						<MenuItem
							className={
								classNames(
									NavBar.classNames.navItem,
									item.classNames,
									{
										active: isSelected
									}
								)
							}
							key={item.value}
							icon={icon}
							isSelected={isSelected}
							shortcut={item.shortcut}
							onClick={() => {
								if (!isSelected) {
									props.onNavigate(item.value);
								}
							}}
						>
							{item.label}
						</MenuItem>
					);
				})}

			</NavigableMenu>
		</div>
	);
}

NavBar.propTypes = {
	onNavigate: PropTypes.func.isRequired,
	choices: PropTypes.arrayOf(
		PropTypes.shape({
			value: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
			shortcut: PropTypes.string,
			classNames: PropTypes.string
		})),
	value: PropTypes.string
};

NavBar.defaultProps = [

	{
		value: 'forms',
		label: 'Forms',
		icon: 'feedback'
	},
	{
		value: 'settings',
		label: 'Settings',
		icon: 'admin-settings'
	}
];

NavBar.classNames = {
	wrapper: [
		'caldera-editor-header', 'caldera-editor-subnav'
	],
	navItem: 'caldera-subnav',
	active: 'active'
}