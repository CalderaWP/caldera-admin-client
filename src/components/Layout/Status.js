import React from 'react';
import {Spinner} from '@wordpress/components';
import PropTypes from 'prop-types'
import {Admin} from '@caldera-labs/components'

export const Status = (props) => {
	if (props.updating) {
		return (
			<Spinner/>
		)
	}
	return (
		<Admin.StatusIndicator
			message={props.message}
			show={props.show}
			success={props.success}
		/>
	)
};

Status.propTypes = {
	updating: PropTypes.bool,
	show: PropTypes.bool,
	success: PropTypes.bool,
	message: PropTypes.string,
};

Status.defaultProps = {
	show: false,
	success: true,
};


