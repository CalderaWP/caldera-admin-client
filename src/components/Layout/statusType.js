import PropTypes from "prop-types";
export const statusShape = {
	message: PropTypes.string,
	show: PropTypes.bool,
	success: PropTypes.bool,
	updating: PropTypes.bool
};
/**
 * Prop types for Status component that shows message and/or spinner
 */
export default PropTypes.shape(statusShape);