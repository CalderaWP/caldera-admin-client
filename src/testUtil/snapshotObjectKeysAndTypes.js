/**
 * Test utility to snapshot an objects enumerable keys and their type
 * @param object
 */
export function snapshotObjectKeysAndTypes(object) {
	let test = {};
	Object.keys(object).forEach(key => {
		test[key] = typeof object[key];
	});
	expect(JSON.stringify(test)).toMatchSnapshot();
}
