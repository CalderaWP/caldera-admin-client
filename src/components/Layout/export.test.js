import Layout from './index';
import {snapshotObjectKeysAndTypes} from "../../testUtil/snapshotObjectKeysAndTypes";

describe( 'Layout export', () => {
	it( 'index has the right types and keys', () => {
		snapshotObjectKeysAndTypes(Layout);
	});
});