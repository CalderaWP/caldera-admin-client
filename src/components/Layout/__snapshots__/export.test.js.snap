// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout export index has the right types and keys 1`] = `"{\\"Status\\":\\"function\\"}"`;
